### ISSUE TITLE
Time limit not respected when still in first solution heuristic

### ISSUE THREAD

**Issue by KeremAslan** [google/or-tools#2212]
**What version of OR-tools and what language are you using?**
Version: v7.7.7810
Language: Java

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
Routing

**What operating system (Linux, Windows, ...) and version?**
MacOS

It seems that OR-Tools doesn't respect any time limit if it is stuck in the initial solution heuristic. I don't observe the same behaviour in the Python wrapper. I have set a time limit as follows
```kotlin
main
.defaultRoutingSearchParameters()
.toBuilder()
.setLogSearch(true)                
.setFirstSolutionStrategy(FirstSolutionStrategy.Value.PARALLEL_CHEAPEST_INSERTION) 
.setLocalSearchMetaheuristic(LocalSearchMetaheuristic.Value.GUIDED_LOCAL_SEARCH)
.setTimeLimit(OrToolsDuration.newBuilder().setSeconds(1200)
.build())
```

Then I use the same search configuration on a bunch of datasets and this is what I get when the solver can't find any solutions:

```
I1104 04:24:54.402680 207704064 routing.cc:2975] All Unperformed Solution (29200000, time = 56 ms, memory used = 743.14 MB)
I1104 04:24:54.402715 207704064 search.cc:254] Start search (memory used = 743.14 MB)
I1104 04:24:54.407321 207704064 search.cc:254] Root node processed (time = 4 ms, constraints = 6121, memory used = 743.45 MB)
I1104 06:25:49.887532 207704064 search.cc:254] Finished search tree (time = 7255485 ms, branches = 108, failures = 17372, neighbors = 38780, filtered neighbors = 17298, accepted neigbors = 2, memory used = 749.30 MB)
I1104 06:25:49.889379 207704064 search.cc:254] End search (time = 7255487 ms, branches = 108, failures = 17372, memory used = 749.30 MB, speed = 0 branches/s)
[INFO ] 2020-11-04 06:25:49.889 [main] PdvrpSolver - OR-tools Solver finished in  7255.543 seconds
```

If however, it can find solutions it does respect the time limit. 
```
I1104 08:30:03.889158 207704064 routing.cc:2975] All Unperformed Solution (22600000, time = 63 ms, memory used = 757.62 MB)
I1104 08:30:03.889204 207704064 search.cc:254] Start search (memory used = 757.62 MB)
I1104 08:30:03.892608 207704064 search.cc:254] Root node processed (time = 3 ms, constraints = 5065, memory used = 757.84 MB)
I1104 08:38:36.343933 207704064 search.cc:254] Solution #0 (1601575, time = 512454 ms, branches = 1238, failures = 510472, depth = 33, neighbors = 8474096, filtered neighbors = 510154, accepted neighbors = 286, memory used = 761.16 MB, limit = 42%)
.....
I1104 09:10:03.648778 207704064 search.cc:254] Solution #161 (200947, objective maximum = 201845, time = 1194395 ms, branches = 2474, failures = 724498, depth = 33, PairExchangeOperator, neighbors = 80036513, filtered neighbors = 723749, accepted neighbors = 536, memory used = 179.77 MB, limit = 99%)
I1104 09:10:09.177457 207704064 search.cc:254] Finished search tree (time = 1199924 ms, branches = 2478, failures = 724529, neighbors = 81043372, filtered neighbors = 723749, accepted neigbors = 536, memory used = 179.77 MB)
I1104 09:10:09.179764 207704064 search.cc:254] End search (time = 1199926 ms, branches = 2478, failures = 724529, memory used = 180.13 MB, speed = 2 branches/s)
[INFO ] 2020-11-04 09:10:09.181 [main] PdvrpSolver - OR-tools Solver finished in  1200.02 seconds
```

**Comment by Mizux** [comment]
Does the time limit only involved during the local search ? need to double check...

**Comment by KeremAslan** [comment]
Yeah, it only worked during the local search phase.

**Comment by Mizux** [comment]
I don't think someone is (or will) working on preemptive stop during the initial solution search.

**Comment by KeremAslan** [comment]
Hmm ok. 

For anyone wondering how to get around this. Using the first solution strategy `ALL_UNPERFORMED` seems to not have this problem (as it skips building a solution and unassigns everything).

**Comment by Woodz** [comment]
@Mizux please note that this functionality is not documented and does not match with the current documentation that implies that the time limit applies to the entire search, not to specific phases. I strongly encourage this to be classified as a bug and documented until it is fixed
![image](https://user-images.githubusercontent.com/966157/100404677-ed590e00-309c-11eb-9215-944df7a7f02b.png)

**Comment by EricPrescottGagnon** [comment]
@Mizux Hello, I am running into this issue as well. I understand it is not a priority, but why was the issue closed?

**Comment by Mizux** [comment]
let's open it to track doc update

**Comment by lperron** [comment]
I think the first solution code was completely rewritten after 7.7.

So there will not be any fix on the 7.7 code.

Closing
